app/src/main/java/org/kiwix/kiwixmobile/main/ConfirmationAlertDialogFragment.java

```diff
@@ -0,0 +1,85 @@
+package org.kiwix.kiwixmobile.main;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.Fragment;
+
+import org.kiwix.kiwixmobile.KiwixApplication;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
+
+import javax.inject.Inject;
+
+/**
+ * Created by @Aditya-Sood as a part of GSoC 2019
+ *
+ * This is a generic helper class for displaying a 2-button (positive & negative) confirmation dialog fragment on top of an existing dialog fragment
+ *  - Only for confirmation dialogs with a Positive & Negative button
+ *  - If you also need a Neutral button, add it selectively (if-else) for the required use case (Take care of the callback interface as well)
+ *
+ * Currently used as:
+ *  - Helper class to show the alert dialog in case the user tries to exit the {@link AddNoteDialog} with unsaved file changes
+ *
+ **/
+
+public class ConfirmationAlertDialogFragment extends DialogFragment {
+
+  public static String TAG = "ConfirmationAlertDialog";
+  @Inject SharedPreferenceUtil sharedPreferenceUtil;
+  private int stringResource;
+  private String parentDialogFragmentTAG;
+
+  public ConfirmationAlertDialogFragment(String dialogFragmentTAG, int resourceId) {
+    super();
+    parentDialogFragmentTAG = dialogFragmentTAG;
+    stringResource = resourceId;
+    KiwixApplication.getApplicationComponent().inject(this);
```

Comment:
On the topic of dependency injection there are largely 2 types, field injection and constructor injection.
Field Injection is necessary when we do not control the creation of these objects, a prime example is an activity.
Constructor Injection is the preferred method as it creates much more easily testable classes without ever involving a DI framework.

Here the field injection is not required because you control this object's construction, just pass in the sharedPreferenceUtil to the constructor.

As an aside on this constructor I would need a very good excuse to have my parameters not match my field names. I would change the parameters or the names.

The super call here is also redundant

Reply:
> On the topic of dependency injection there are largely 2 types, field injection and constructor injection.
> Field Injection is necessary when we do not control the creation of these objects, a prime example is an activity.
> Constructor Injection is the preferred method as it creates much more easily testable classes without ever involving a DI framework.
> 
> Here the field injection is not required because you control this object's construction, just pass in the sharedPreferenceUtil to the constructor.

Thanks, I'll do this right away
> As an aside on this constructor I would need a very good excuse to have my parameters not match my field names. I would change the parameters or the fields.
> 
Will fix this

> The super call here is also redundant

Since ConfirmationAlertDialogFragment extends from DialogFragment, wouldn't I have to call the base class constructor as well?

Reply:
A no args super call will always be implicitly called.
![image](https://user-images.githubusercontent.com/3358282/59858049-205c8580-9372-11e9-8f10-57b71c967177.png)

Reply:
Thanks, will get to it

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1198/commits/5723554c53b11c0fd28f987cdb91962f01ee634d)